js/src/api/transport/ws/ws.js

```diff
@@ -249,6 +292,38 @@ export default class Ws extends JsonRpcBase {
     });
   }
 
+  subscribe (method, callback, ...params) {
+    return new Promise((resolve, reject) => {
+      const id = this.id;
+      const json = this.encode(method, params);
+      const subscription = true;
+
+      this._messages[id] = { id, method, params, json, resolve, reject, callback, subscription };
+
+      this._send(id);
+    });
+  }
+
+  unsubscribe (messageId) {
+    return new Promise(() => {
```

Comment:
Why this promise is never resolved? I think it should be resolved in `this._messages[id]`, no?